common/chains/client/types.go

```diff
@@ -0,0 +1,130 @@
+package client
+
+import (
+	"context"
+	"math/big"
+
+	feetypes "github.com/smartcontractkit/chainlink/v2/common/fee/types"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+	"github.com/smartcontractkit/chainlink/v2/core/assets"
+	"github.com/smartcontractkit/chainlink/v2/core/utils"
+)
+
+// RPC includes all the necessary methods for a multi-node client to interact directly with any RPC endpoint.
+type RPC[
+	CHAIN_ID types.ID,
+	SEQ types.Sequence,
+	ADDR types.Hashable,
+	BLOCK_HASH types.Hashable,
+	TX any,
+	TX_HASH types.Hashable,
+	EVENT any,
+	EVENT_OPS any,
+	TX_RECEIPT types.Receipt[TX_HASH, BLOCK_HASH],
+	FEE feetypes.Fee,
+	HEAD types.Head[BLOCK_HASH],
+
+] interface {
+	NodeClient[
+		CHAIN_ID,
+		HEAD,
+	]
+	ClientAPI[
+		CHAIN_ID,
+		SEQ,
+		ADDR,
+		BLOCK_HASH,
+		TX,
+		TX_HASH,
+		EVENT,
+		EVENT_OPS,
+		TX_RECEIPT,
+		FEE,
+		HEAD,
+	]
+}
+
+// Head is the interface required by the NodeClient
+type Head interface {
+	BlockNumber() int64
+	BlockDifficulty() *utils.Big
+}
+
+// NodeClient includes all the necessary RPC methods required by a node.
+type NodeClient[
```

Comment:
Is this NodeClient ever used outside of this pckage?
I think not. Could we unexport it, call it nodeClient, and move it out from this file?